pytorch_lightning/utilities/auto_restart.py

```diff
@@ -737,3 +739,15 @@ def _teardown_dataloader_get_iterators() -> None:
     if get_iterator:
         DataLoader._get_iterator = get_iterator
         del DataLoader._ori_get_iterator
+
+
+def _collect_states_on_rank_zero_over_collection(state_dict: Any, device: torch.device) -> Any:
+    """This utility collects the state across processes for a collection of state."""
+
+    def fn(state: Dict):
+        nonlocal device
+        if state.get("state", None) is not None:
```

Comment:
```suggestion
        if "state" in state:
```
 or not?

Reply:
I believe the function should document that it will only collect dictionaries with a key "state" in it.

Reply:
Done !

Reply:
I believed you missed my comment here.
You also changed `_collect_states_on_rank_zero` so this code might be outdated.

Reply:
Done !